### PR TITLE
implement get_variable_names() and get_variable_value()

### PR DESCRIPTION
**Pull request by kevinykuo**
- Add `variable_names()` and `variable_value()` to extract names and values of variables from estimator models to replace `coef.tf_estimator()`

These are wrappers around https://www.tensorflow.org/api_docs/python/tf/estimator/Estimator#get_variable_names with a couple enhancements in `variable_value()`:
1. User can specify multiple variables to extract by passing a vector to `variable`.
2. When `variable` is not specified, return all variables in the graph. See unit tests for usage.

(closes https://github.com/rstudio/tfestimators/issues/121)

### CODE REVIEW

**Comment by jjallaire** [comment]
How about we remove the `get_` prefix?

**Comment by kevinykuo** [comment]
@jjallaire I like that, was debating with myself since we're mostly faithful to Python naming with other stuff. Will make change.

**Comment by kevinykuo** [comment]
(I see that travis failed for TF 1.3; will fix.)

File: R/tf_estimator.R [hunk 1]
```diff
@@ -376,25 +376,39 @@ export_savedmodel.tf_estimator <- function(object,
...
   }
-  ckp <- training_lib$checkpoint_utils$load_checkpoint(latest_checkpoint(model_dir))
-  var_names <- list_variable_names(model_dir)
-  cleaned_vars <- lapply(var_names, function(var_name) ckp$get_tensor(var_name[[1]]))
-  names(cleaned_vars) <- var_names
-  cleaned_vars
+  
+  variable %>%
+    lapply(object$estimator$get_variable_value) %>%
+    rlang::set_names(unlist(variable))
```
Comment (kevinushey) [inline R/tf_estimator.R]: Is it safe to unlist here? What if the value of a weight / variable is itself a list (could that occur in practice)?

**Comment by terrytangyuan** [comment]
Looks great!

**Pull request merged**